app/code/Magento/StoreGraphQl/Model/Resolver/Store/StoreConfigDataProvider.php

```diff
@@ -55,24 +56,36 @@ public function __construct(
      */
     public function getStoreConfigData(StoreInterface $store): array
     {
-        $storeConfigData = array_merge(
-            $this->getBaseConfigData($store),
-            $this->getExtendedConfigData((int)$store->getId())
-        );
-        return $storeConfigData;
+        $defaultStoreConfig = $this->storeConfigManager->getStoreConfig($store);
+        return $this->prepareStoreConfigData($defaultStoreConfig);
     }
 
     /**
-     * Get base config data
+     * Get available stores
      *
-     * @param StoreInterface $store
      * @return array
      */
-    private function getBaseConfigData(StoreInterface $store) : array
+    public function getAvailableStores(): array
     {
-        $storeConfig = current($this->storeConfigManager->getStoreConfigs([$store->getCode()]));
+        $storesConfigData = [];
+        $storeConfigs = $this->storeConfigManager->getStoreConfigs();
```

Comment:
We should only return available stores for the current website. This is returning all stores for all websites